airflow/providers/google/cloud/transfers/gdrive_to_local.py

```diff
@@ -0,0 +1,96 @@
+    """
+    Writes a Google Drive file into local Storage.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GoogleDriveToGCSOperator`
+
+    :param output_file: Path to downloaded file
+    :type output_file: str
+    :param folder_id: The folder id of the folder in which the Google Drive file resides
+    :type folder_id: str
+    :param filename: The name of the file residing in Google Drive
+    :type filename: str
+    :param drive_id: Optional. The id of the shared Google Drive in which the file resides.
+    :type drive_id: str
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = [
+        "output_file",
+        "folder_id",
+        "filename",
+        "drive_id",
+        "impersonation_chain",
+    ]
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        output_file: str,
+        filename: str,
+        folder_id: str,
+        drive_id: Optional[str] = None,
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.output_file = output_file
+        self.folder_id = folder_id
+        self.drive_id = drive_id
+        self.filename = filename
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+        self.file_metadata = None
+
+    def execute(self, context):
+        self.log.info('Executing download: %s into %s', self.filename, self.output_file)
+        gdrive_hook = GoogleDriveHook(
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.impersonation_chain,
+        )
+        self.file_metadata = gdrive_hook.get_file_id(
+            folder_id=self.folder_id, file_name=self.filename, drive_id=self.drive_id
+        )
+
+        with open(self.output_file, "wb") as file:
+            gdrive_hook.download_file(file_id=self.file_metadata["id"], file_handle=file)
```

Comment:
```suggestion


    def execute(self, context):
        self.log.info('Executing download: %s into %s', self.filename, self.output_file)
        gdrive_hook = GoogleDriveHook(
            delegate_to=self.delegate_to,
            impersonation_chain=self.impersonation_chain,
        )
        file_metadata = gdrive_hook.get_file_id(
            folder_id=self.folder_id, file_name=self.filename, drive_id=self.drive_id
        )

        with open(self.output_file, "wb") as file:
            gdrive_hook.download_file(file_id=file_metadata["id"], file_handle=file)
```
Probably no need to keep this as instance attribute